feat: cloudant da (#32)
terraform-ibm-modules/terraform-ibm-cloudant: solutions/dedicated/README.md

# IBM Cloudant Dedicated on IBM Cloud Dedicated

> This architecture creates an IBM Cloudant Dedicated instance in IBM Cloud Dedicated and provisions the following resources:

Reviewer: I think this needs to be re-worded:
`provisions the following resources` -> `supports provisioning the following resources` otherwise it seems as if the DA will create a database by default, which is not the case (its an opt in feature)

# IBM Cloudant Dedicated on IBM Cloud Dedicated

> This architecture creates an IBM Cloudant Dedicated instance in IBM Cloud Dedicated and supports provisioning the following resources:

Reviewer: I think this reads better: `in an IBM Cloud Dedicated environment`

Author: Can we leave it as is because not everything is in the dedicated environment - like resource group

Reviewer: I was suggesting: 

This architecture creates an IBM Cloudant Dedicated instance in an IBM Cloud Dedicated environment and supports provisioning the following resources

- A resource group, if one is not passed in.
- An IBM Cloudant Dedicated instance on IBM Cloud Dedicated
- A IBM Cloudant database

> ![dedicated-cloudant](../../images/dedicated_cloudant.svg)

Reviewer: Can you add the diagram into a folder called `reference-architectures` instead. This is where they will need to be for when we publish a public DA.

> # Deployable Architecture (DA) for the dedicated IBM Cloudant instance

Author: As per the docs:

> IBM Cloud Dedicated customers can also choose to use IBM Cloudant in IBM Cloud Public by using the syndicated catalog, including the Lite and Standard plan instances. IBM Cloud Dedicated users aren't eligible to buy the Dedicated Hardware plan from the IBM Cloud Public catalog. IBM Cloud Dedicated users must use IBM Cloudant Dedicated on IBM Cloud Dedicated if dedicated hardware is required.

I think the exact title should be `IBM Cloudant Dedicated on IBM Cloud Dedicated` as it is deployed on the dedicated hardware. I have changed the title to that. Thoughts? 

@ocofaigh @SirSpidey

> # Deployable Architecture (DA) for the dedicated IBM Cloudant instance

Reviewer 2: It isn't a pretty title...
If we're not expecting another module of IBM Cloudant Dedicated on `something else`, I think `IBM Cloudant Dedicated` is good enough as the title. We can get into the ICD on ICD mesh of words in the description.

> Deployable Architecture for the dedicated IBM Cloudant instance which deploys the following resources:

Reviewer: which supports provisioning the following resources:

- A resource group, if one is not passed in.

> - A dedicated cloudant instance

Reviewer: An IBM Cloudant Dedicated instance in IBM Cloud Dedicated (seems strange, but thats what Cloudant [docs](https://cloud.ibm.com/docs/Cloudant?topic=Cloudant-ibm-cloud-dedicated) have). @SirSpidey can advise

> - A basic database in the Cloudant instance

Reviewer: By default it doesn't create any database. It supports optionally creating one, so suggest saying:

`- A Cloudant database`

> 

Reviewer: I think we need to add a Before you begin section:

```
## Before you begin

* You need an [IBM Cloudant Dedicated](https://cloud.ibm.com/docs/Cloudant?topic=Cloudant-ibm-cloud-dedicated) Hardware plan instance in order to provision the Cloudant instance in
```

# IBM Cloudant Dedicated on IBM Cloud Dedicated

> Solutions for an IBM Cloudant dedicated instance which supports provisioning the following resources:

Reviewer 2: ```suggestion
This architecture creates an IBM Cloudant Dedicated instance in IBM Cloud Dedicated and provisions the following resources:
```

- A resource group, if one is not passed in.

> - An IBM Cloudant Dedicated instance in IBM Cloud Dedicated

Reviewer 2: ```suggestion
- An IBM Cloudant Dedicated instance on IBM Cloud Dedicated
```

> - A Cloudant database

Reviewer 2: ```suggestion
- A IBM Cloudant database
```

## Before you begin

> * You need an [IBM Cloudant Dedicated](https://cloud.ibm.com/docs/Cloudant?topic=Cloudant-ibm-cloud-dedicated) Hardware plan instance in order to provision the Cloudant instance in.

Reviewer 2: ```suggestion
* You need an IBM Cloudant [Dedicated Hardware plan instance](https://cloud.ibm.com/docs/Cloudant?topic=Cloudant-ibm-cloud-dedicated) for the IBM Cloudant instance.
```

Using a Dedicated Hardware plan instance

> <!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->

Reviewer 2: This infor will appear with the DA in the cloud catalog, so we want to make sure that it makes sense in that context.

Author: Removing the pre-commit-terraform docs hook, we don't want to display this information (Same as in SLZ module).
